sound/soc/intel/boards/sof_sdw.c

```diff
@@ -411,25 +411,30 @@ static int create_codec_dai_name(struct device *dev,
 
 static int set_codec_init_func(const struct snd_soc_acpi_link_adr *link,
 			       struct snd_soc_dai_link *dai_links,
-			       bool playback)
+			       bool playback, int group_id)
 {
 	int i;
 
-	for (i = 0; i < link->num_adr; i++) {
-		unsigned int part_id;
-		int codec_index;
-
-		part_id = SDW_PART_ID(link->adr_d[i].adr);
-		codec_index = find_codec_info_part(part_id);
+	do {
```

Comment:
Took me a while to figure out why we need to loop here. It looks like now with aggregation the loop over links in `sof_card_dai_links_create()` only calls `create_sdw_dailink()` for the first link in each group. That's why you now have to walk over all links, looking for others from the same group. Maybe you could add a comment about that for the next reader :-)